### PR TITLE
Add environment variables

### PR DESCRIPTION
**Pull request by 0phelia36**
## TL;DR
Added ability to configure most variables in the configuration file using environment variables. 

Also used these to configure the docker worker to fix a bug where when tasks were completed they weren't signalling the API to move on to the next task since it was trying to go to localhost:8081 instead of api:8081. This only occurred if the web gui wasn't open since the GUI polling for status updates also triggers moving on to the next task.

## What
What is affected by this PR?
- [ ] API
- [ ] GUI
- [ ] Hardware Integration
- [X] OS/Deployment
- [ ] Documentation
- [ ] Other (please describe in notes)

## Testing
How was this tested?
- [ ] Locally Virtualized
- [ ] Raspberry Pi
- [ ] With Hardware
- [ ] Other (please describe in notes)

## Notes
How was your day?

### CODE REVIEW

**Review by 0x42red — APPROVED** [review]
Nice let's kill hardcoded vars
